Score

hwchen opened the pull request.

hwchen committed:
Add sos and sodos to mm_score

- change mm_score from int to list of int: [score, sos, sodos]
- update Player repr to show mm_score list
- update Tournament.calculate_mm_score() to calculate sos and sodos
- delete Tournament.generate_candidate_pairings() (dead code)
- update Tournament.pairing_score() to use mm_score list
- update tournament.yaml to use mm_score list

hwchen committed:
update tests

hwchen committed:
clean data files

hndrewaall commented on mcmahon.py:
```diff
@@ -131,8 +146,8 @@ def pairing_score(self, player_list):
         while temp_list:
-            score += abs(self.players[temp_list.pop()].mm_score -
-                         self.players[temp_list.pop()].mm_score)
+            score += abs(self.players[temp_list.pop()].mm_score[0] -
```
I don't think this is good. Ideally we would have a class that defines not just ordering but also the difference between two scores.

hndrewaall commented on mcmahon.py:
```diff
@@ -215,7 +210,7 @@ def generate_pairing(self, sample_size):
         match = self.rounds[round_][board]
         match.winner = winner
-        self.players[winner].mm_score += 1
+        self.players[winner].mm_score[0] += 1
```
At this point, is there even any reason to do this? It leaves SOS and SODOS incorrect, and we basically assume we are going to manually trigger a recomputation down the line.

hndrewaall commented on mcmahon.py:
```diff
@@ -347,28 +343,30 @@ def test_winner(self):
+                   Player('David',     -4, 5723,  [1,0,0], 1, 2),
+                   Player('Eric',      -4, 5723,  [1,0,0], 1, 2),
+                   Player('Howie',     -4, 5723,  [1,0,0], 1, 2),
                    ]
```
:stuck_out_tongue_closed_eyes:

hndrewaall commented on mcmahon.py:
```diff
@@ -347,28 +343,30 @@ def test_winner(self):
                    ]
         self.tournament = Tournament.new_tournament(players)
+        for player in self.tournament.players:
+            print(yaml.dump(self.tournament))
```
Print statements like this shouldn't be committed as part of the unittest (even though they are super helpful while developing).

hwchen committed:
Improve formatting of standings and pairings

- Improve formatting of Tournament.wall_list()
- Improve formatting of Tournament.pairings_list()

hndrewaall commented:
good. whatever. AHHHH
